src/main/java/language/LanguageCommand.java

```diff
@@ -0,0 +1,383 @@
+                                        "name",
+                                        "Name",
+                                        true
+                                )
+                        )),
+                new SlashCommandOptionBuilder()
+                        .setType(SlashCommandOptionType.SUB_COMMAND)
+                        .setName("query")
+                        .setDescription("Displays information on a language, and its immediate relatives.")
+                        .setOptions(Lists.newArrayList(
+                                SlashCommandOption.create(
+                                        SlashCommandOptionType.STRING,
+                                        "name",
+                                        "Name",
+                                        true
+                                )
+                        )),
+                new SlashCommandOptionBuilder()
+                        .setType(SlashCommandOptionType.SUB_COMMAND)
+                        .setName("connect")
+                        .setDescription("Connect Languages")
+                        .setOptions(Lists.newArrayList(
+                                SlashCommandOption.create(
+                                        SlashCommandOptionType.STRING,
+                                        "language1",
+                                        "First Language",
+                                        true
+                                ),
+                                SlashCommandOption.create(
+                                        SlashCommandOptionType.STRING,
+                                        "language2",
+                                        "Second Language",
+                                        true
+                                )
+                        )),
+                new SlashCommandOptionBuilder()
+                        .setType(SlashCommandOptionType.SUB_COMMAND)
+                        .setName("calculate")
+                        .setDescription("Calculates the difficulty of linguistics checks, will use User as default input")
+                        .setOptions(Lists.newArrayList(
+                                SlashCommandOption.create(
+                                        SlashCommandOptionType.STRING,
+                                        "target",
+                                        "The Language to Target",
+                                        true
+                                ),
+                                SlashCommandOption.create(
+                                        SlashCommandOptionType.MENTIONABLE,
+                                        "characters",
+                                        "The Character or Party to pull languages from as input",
+                                        false
+                                ),
+                                SlashCommandOption.create(
```

Comment:
I would break this into 2 commands by creating a subcommand group, one for pulling from the sheet and one for manual entry. What happens if you define both a user and a list of languages?